_delphi_utils_python/tests/test_archive.py

```diff
@@ -187,47 +238,37 @@ def test_diff_and_filter_exports(self, tmp_path):
             arch_diff.diff_exports()
 
         # Simulate cache updated, and signal ran finish
-        for csv_name, df in CSVS_BEFORE.items():
-            df.to_csv(join(cache_dir, f"{csv_name}.csv"), index=False)
-        for csv_name, df in CSVS_AFTER.items():
-            df.to_csv(join(export_dir, f"{csv_name}.csv"), index=False)
+        for csv_name, dfs in CSVS.items():
+            if dfs.before is not None:
+                dfs.before.to_csv(join(cache_dir, f"{csv_name}.csv"), index=False)
+            if dfs.after is not None:
+                dfs.after.to_csv(join(export_dir, f"{csv_name}.csv"), index=False)
         arch_diff._cache_updated = True
 
         deleted_files, common_diffs, new_files = arch_diff.diff_exports()
 
         # Check return values
```

Comment:
Thinking this comment is worth updating. 

```suggestion
        # Check deleted, common, diffed, and new file names match expected
```